plugin/sycl/tree/updater_quantile_hist.cc

```diff
@@ -27,21 +29,115 @@ void QuantileHistMaker::Configure(const Args& args) {
   const DeviceOrd device_spec = ctx_->Device();
   qu_ = device_manager.GetQueue(device_spec);
 
+  // initialize pruner
+  if (!pruner_) {
+    pruner_.reset(TreeUpdater::Create("prune", ctx_, task_));
```

Comment:
The pruner is not using sycl, do we need it? The existing hist tree updater doesn't use the pruner. Instead, it checks whether a node is splittable before growing, it's not only more efficient due to less computation, but also simplier.

Reply:
Make sense. I have removed pruner from sycl.